pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -73,6 +97,31 @@ func (pl *ImageLocality) ScoreExtensions() framework.ScoreExtensions {
 	return nil
 }
 
+// Clone implements the mandatory Clone interface. We don't really copy the data since
+// there is no need for that.
+func (s *preScoreState) Clone() framework.StateData {
+	return s
+}
+
+func (pl *ImageLocality) PreScore(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodes []*v1.Node) *framework.Status {
+	var containersPullAlways []v1.Container
+
+	// Filtering containers with ImagePullPolicy different from Always
+	// Ones with Always will score 0 point either way
+	for _, container := range pod.Spec.Containers {
+		if container.ImagePullPolicy != v1.PullAlways {
+			containersPullAlways = append(containersPullAlways, container)
```

Comment:
Up to now, `container.Image` is enough, let's use a `sets.String` instead which helps to reduce the memory allocation?